swap/swap.go

```diff
@@ -344,6 +368,42 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 	return err
 }
 
+func (s *Swap) handleConfirmChequeMsg(ctx context.Context, p *Peer, msg *ConfirmChequeMsg) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	cheque := msg.Cheque
+
+	if p.getPendingCheque() == nil {
+		p.logger.Warn("ignoring confirm msg, no pending cheque")
```

Comment:
1. You're right, I missed a return here.
2. In the current implementation this can happen. If there is further swap traffic while waiting for the confirmation the pending cheque might be resent. In that case `handleEmitChequeMsg` will run twice, only updating its state the first time but responding with a confirmation both times. When `handleConfirmChequeMsg` runs for the 2nd time `pendingCheque` will be `nil`. Ideally `Add` should not try to resend the `pendingCheque` within some timeout but this is not implemented in this PR.

Reply:
ok. maybe log the cheque serial number in the logline too? some identifying details?